bin/oneoff/add_progress_v2_closed_beta_users.rb

```diff
@@ -4,16 +4,23 @@
 require 'csv'
 
 # Run using `ruby add_progress_v2_closed_beta_users.rb <path_to_csv>`
-# For a given csv file, find all users by email and update their `progress_table_v2_closed_beta` user preference to true.
+# For a given csv file, find all users by id and update their `progress_table_v2_closed_beta` user preference to true.
 
-# CSV file must have a `email` column
+# CSV file must have a `id` column
 
-email_csv = ARGV[0]
-unless email_csv
-  puts 'Usage: add_progress_v2_closed_beta_users <email_csv>'
+teacher_id_csv = ARGV[0]
+unless teacher_id_csv
+  puts 'Usage: add_progress_v2_closed_beta_users <teacher_id_csv>'
   exit
 end
 
-CSV.table(email_csv).each do |row|
-  User.where(email: row[:email]).map {|u| u.update!(progress_table_v2_closed_beta: true)}
+count = 0
+CSV.table(teacher_id_csv).each do |row|
+  user = User.find(id: row[:id])
```

Comment:
I believe `find` always takes an ID or list of IDs, while `find_by` is what you would use with named fields. Also `find` raises an exception if the record isn't found, which you should probably handle here.

```suggestion
  user = User.find(row[:id])
```